tket/src/Ops/ClassicalOps.cpp

```diff
@@ -239,29 +240,38 @@ std::vector<bool> ClassicalTransformOp::eval(const std::vector<bool> &x) const {
 }
 
 WASMOp::WASMOp(
-    unsigned _n, std::vector<unsigned> _ni_vec, std::vector<unsigned> _no_vec,
-    const std::string &_func_name, const std::string &_wasm_uid)
+    unsigned _n, unsigned _ww_n, std::vector<unsigned> _width_i_parameter,
+    std::vector<unsigned> _width_o_parameter, const std::string &_func_name,
+    const std::string &_wasm_file_uid)
     : ClassicalOp(
           OpType::WASM,
           std::accumulate(
-              _ni_vec.begin(), _ni_vec.end(), decltype(_ni_vec)::value_type(0)),
+              _width_i_parameter.begin(), _width_i_parameter.end(),
+              (unsigned)0),
           0,
           std::accumulate(
-              _no_vec.begin(), _no_vec.end(), decltype(_no_vec)::value_type(0)),
+              _width_o_parameter.begin(), _width_o_parameter.end(),
+              (unsigned)0),
           "WASM"),
       n_(_n),
-      n_i32_(_ni_vec.size() + _no_vec.size()),
-      ni_vec_(_ni_vec),
-      no_vec_(_no_vec),
+      ww_n_(_ww_n),
+      n_i32_(_width_i_parameter.size() + _width_o_parameter.size()),
+      width_i_parameter_(_width_i_parameter),
+      width_o_parameter_(_width_o_parameter),
       func_name_(_func_name),
-      wasm_uid_(_wasm_uid) {
+      wasm_file_uid_(_wasm_file_uid) {
   unsigned sum_of_i32 =
       std::accumulate(
-          ni_vec_.begin(), ni_vec_.end(), decltype(ni_vec_)::value_type(0)) +
+          width_i_parameter_.begin(), width_i_parameter_.end(), (unsigned)0) +
       std::accumulate(
-          no_vec_.begin(), no_vec_.end(), decltype(no_vec_)::value_type(0));
+          width_o_parameter_.begin(), width_o_parameter_.end(), (unsigned)0);
```

Comment:
Is the cast needed?

Reply:
No, removed in @cqc-melf
[remove cast from sum in parameters](https://github.com/CQCL/tket/pull/743/commits/d27656e18fba7fccb2a6dd88f4014111ccf06214)
[d27656e](https://github.com/CQCL/tket/pull/743/commits/d27656e18fba7fccb2a6dd88f4014111ccf06214)